### PR TITLE
[wanda] add workrepo ci caching support

### PR DESCRIPTION
**Pull request by aslonnie**
will directly retag the image on cache hit.

### CODE REVIEW

**Comment by aslonnie** [comment]
10s build on cache hit: https://buildkite.com/ray-project/rayci/builds/329#018a0723-7267-4657-b16d-b9fdb53a2d07/111-113

half of the time is for installing/building `wanda`, so can be even faster if we download the binary.

**Comment by aslonnie** [comment]
10s ml base building: https://buildkite.com/ray-project/postmerge/builds/154#018a0905-8847-4888-a42d-d172220f8530

File: wanda/forge.go [hunk 1]
```diff
@@ -161,26 +181,60 @@ func (f *Forge) Build(spec *Spec) error {
...
+		}
+
+		wt, err := cranename.NewTag(workTag)
+		if err != nil {
+			return fmt.Errorf("parse work tag %q: %w", workTag, err)
+		}
+
+		desc, err := remote.Get(ct, f.remoteOpts...)
+		if err != nil {
+			log.Printf("fetch cache image: %v", err)
```
Comment (can-anyscale) [inline wanda/forge.go:202]: the log should say 'failed to fetch cache image' or 'no cache image found'?

**Review by can-anyscale — APPROVED** [review]
super!

**Comment by can-anyscale** [comment]
do we need to invalidate the cache once a while to upgrade apt-get?

**Comment by aslonnie** [comment]
> Member

I will add that later. the ECR also has a 14-day retention period.

File: wanda/forge.go (hunk 1 above)
Comment (aslonnie) [reply wanda/forge.go]: changed to "Cache image miss"

**Pull request merged**